apps/opik-documentation/documentation/docs/cookbook/bedrock.ipynb

```diff
@@ -0,0 +1,307 @@
+    "        modelId=model_id,\n",
+    "        messages=messages,\n",
+    "        system=system_prompts,\n",
+    "        inferenceConfig=inference_config,\n",
+    "    )\n",
+    "\n",
+    "    stream = response.get(\"stream\")\n",
+    "    if stream:\n",
+    "        for event in stream:\n",
+    "            if \"messageStart\" in event:\n",
+    "                print(f\"\\nRole: {event['messageStart']['role']}\")\n",
+    "\n",
+    "            if \"contentBlockDelta\" in event:\n",
+    "                print(event[\"contentBlockDelta\"][\"delta\"][\"text\"], end=\"\")\n",
+    "\n",
+    "            if \"messageStop\" in event:\n",
+    "                print(f\"\\nStop reason: {event['messageStop']['stopReason']}\")\n",
+    "\n",
+    "            if \"metadata\" in event:\n",
+    "                metadata = event[\"metadata\"]\n",
+    "                if \"usage\" in metadata:\n",
+    "                    print(\"\\nToken usage\")\n",
+    "                    print(f\"Input tokens: {metadata['usage']['inputTokens']}\")\n",
+    "                    print(f\":Output tokens: {metadata['usage']['outputTokens']}\")\n",
+    "                    print(f\":Total tokens: {metadata['usage']['totalTokens']}\")\n",
+    "                if \"metrics\" in event[\"metadata\"]:\n",
+    "                    print(f\"Latency: {metadata['metrics']['latencyMs']} milliseconds\")\n",
+    "\n",
+    "\n",
+    "system_prompt = \"\"\"You are an app that creates playlists for a radio station\n",
+    "  that plays rock and pop music. Only return song names and the artist.\"\"\"\n",
+    "\n",
+    "# Message to send to the model.\n",
+    "input_text = \"Create a list of 3 pop songs.\"\n",
+    "\n",
+    "\n",
+    "message = {\"role\": \"user\", \"content\": [{\"text\": input_text}]}\n",
+    "messages = [message]\n",
+    "\n",
+    "# System prompts.\n",
+    "system_prompts = [{\"text\": system_prompt}]\n",
+    "\n",
+    "# inference parameters to use.\n",
+    "temperature = 0.5\n",
+    "top_p = 0.9\n",
+    "# Base inference parameters.\n",
+    "inference_config = {\"temperature\": temperature, \"topP\": 0.9}\n",
+    "\n",
+    "\n",
+    "stream_conversation(\n",
+    "    bedrock_client,\n",
+    "    MODEL_ID,\n",
+    "    messages,\n",
+    "    system_prompts,\n",
+    "    inference_config,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Using it with the `track` decorator\n",
+    "\n",
+    "If you have multiple steps in your LLM pipeline, you can use the `track` decorator to log the traces for each step. If Bedrock is called within one of these steps, the LLM call with be associated with that corresponding step:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from opik import track\n",
+    "\n",
+    "os.environ[\"OPIK_PROJECT_NAME\"] = \"openai-integration-demo\"\n",
```

Comment:
Better to pass the project name as part of the track decorator for the top most function